pytorch_lightning/trainer/connectors/callback_connector.py

```diff
@@ -130,8 +130,17 @@ def _configure_swa_callbacks(self):
             self.trainer.callbacks = [StochasticWeightAveraging()] + self.trainer.callbacks
 
     def configure_progress_bar(self, refresh_rate=None, process_position=0):
-        if os.getenv("COLAB_GPU") and refresh_rate is None:
-            # smaller refresh rate on colab causes crashes, choose a higher value
+        use_rich_progress_bar = _RICH_AVAILABLE
```

Comment:
Can we move this code to the top of the function?
```
progress_bars = [c for c in self.trainer.callbacks if isinstance(c, ProgressBarBase)]
        if len(progress_bars) > 1:
            raise MisconfigurationException(
                "You added multiple progress bar callbacks to the Trainer, but currently only one"
                " progress bar is supported."
            )
        if len(progress_bars) == 1:
            progress_bar_callback = progress_bars[0]
```

This function is now getting pretty complex, I think we could make it more understandable by doing three things in this order:
1. If the user has already passed a PBarCallback in "callbacks", just return that immediately
2. Then check if Rich is available and if it is and no RR is set, immediately return a new RichProgressBar()
3. Finally if we get to this point create a TQDMPBar with RR logic as before